scenes/player_2.tscn

```diff
@@ -1,8 +1,68 @@
 [gd_scene load_steps=5 format=3 uid="uid://flgferux1dvg"]
 
-[ext_resource type="Script" path="res://scenes/player.gd" id="1_5vic7"]
 [ext_resource type="Texture2D" uid="uid://bpqevwrcei8il" path="res://assets/p2.png" id="1_8mc2q"]
 
+[sub_resource type="GDScript" id="GDScript_k6tsi"]
+script/source = "extends CharacterBody2D
+
+signal resource_gathered
+signal shoot
+
+@export var speed = 400
+var screen_size
+var resources = 3
+@export var pos = 0
+@export var max_resources = 6
+
+var can_shoot : bool
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	screen_size = get_viewport_rect().size
+	can_shoot = true
+	pass # Replace with function body.
+
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta):
+	var velocity = Input.get_vector(\"move_left\", \"move_right\", \"move_up\", \"move_down\")
+	if velocity.length() > 0:
+		velocity = velocity.normalized() * speed
+		$AnimatedSprite2D.play()
+	else:
+		$AnimatedSprite2D.stop()
+	position += velocity * delta
+	position = position.clamp(Vector2.ZERO, screen_size)
+	
+	if Input.is_action_pressed(\"shoot\") and can_shoot:
+		var direction = Input.get_vector(\"shoot_direction_down\", \"shoot_direction_left\", \"shoot_direction_right\", \"shoot_direction_up\")
+		shoot.emit(position, direction)
+		can_shoot = false
+	pass
+
+func start():
+	position = pos
+	show()
+	$CollisionShape2D.disabled = false
+
+func _on_area_2d_body_entered(body):
+	print(\"test\")
+	if body.name == \"Resource\":
+		
+		var point = body.resource_gathered()
+		print(point)
+		if point and resources < max_resources:
+			print(\"resource detected\")
+			print(resources)
+			resources += 1
+	else:
+		print(body.name)
+	pass # Replace with function body.
+	
+func _on_shot_timer_timeout():
+	can_shoot = true
+"
+
```

Comment:
Not sure why it added this to the script. Any idesa?

Reply:
I'm not entirely sure which part you're referencing. Like you didn't add any of the lines you highlighted?

Reply:
Yeah I didn’t add it.

Reply:
Only part I'm familiar with is "_on_area_2d_body_entered". My guess is that the same player script is on the player_2.tscn, so therefore code added to player_1.tscn may have been replicated across. The "_on_area_2d_body_entered" part is for touching the resource scene. The rest of it looks like stuff from shooting, if it's isn't exactly like the other code you had I couldn't say what it was. I'm down to keep it for now if you are and deal with it later if need be.

Reply:
Yeah let’s keep it for now